Tests/PSKoans.tests.ps1

```diff
@@ -0,0 +1,120 @@
+$here = (Split-Path -Parent $MyInvocation.MyCommand.Path) -replace 'Tests', ''
+$sut = "PSKoans.psm1"
+Import-Module "$here\$sut" -Prefix "TST"
+
+Describe -Name "Initialize-KoanDirectory Unit Testing" -Fixture {
+    Context -Name "Mocked Functions: Test-Path,Remove-Item, and Copy-Item" -Tag @("Mocking") -Fixture {
+        Mock -CommandName Test-Path -MockWith { $false }
+        Mock -CommandName Remove-Item -MockWith {}
+        Mock -CommandName Copy-Item -MockWith {}
+
+        Initialize-TSTKoanDirectory -FirstImport
+
+        It -name "Assert Test-Path mocked" -test {
+            Assert-MockCalled -CommandName Test-Path -Times 1
+        }#It
+        It -name "Assert Remove-Item mocked" -test {
+            Assert-MockCalled -CommandName Remove-Item -Times 0
+        }#It
+        It -name "Assert Copy-Item mocked" -test {
+            Assert-MockCalled -CommandName Copy-Item -Times 1
+        }#It
+    }#Context Mocked Functions
+    Context -Name "Mocked Functions: Remove-Item, and Copy-Item" -Tag @("Mocking") -Fixture {
+        $KoanFolder = $Home | Join-Path -ChildPath 'PSKoans'
+        If( Test-Path $KoanFolder){
+            Remove-Item $KoanFolder -Recurse
+        }
+        Mock -CommandName Remove-Item -MockWith {}
+        Mock -CommandName Copy-Item -MockWith {}
+
+        Initialize-TSTKoanDirectory -FirstImport
+
+        It -name "Assert Remove-Item mocked" -test {
+            Assert-MockCalled -CommandName Remove-Item -Times 0
+        }#It
+        It -name "Assert Copy-Item mocked" -test {
+            Assert-MockCalled -CommandName Copy-Item -Times 1
+        }#It
+    }#Context Mocked Functions
+    Context -Name "No mocking, live execution" -Fixture {
```

Comment:
If we're doing a test with live execution, we should probably remove the created files after the test is complete. Also... we may want to build in some kind of 'debug mode' where it creates the koan directory in `$env:TEMP` instead, so that we don't overwrite anything that already exists when running tests.

Reply:
I debated quite a bit when writing this test. I ultimately agree with your recommendations.